src/Client/ClientBase.cpp

```diff
@@ -572,6 +572,13 @@ try
                 CompressionMethod compression_method = chooseCompressionMethod(out_file, compression_method_string);
                 UInt64 compression_level = 3;
 
+                if (query_with_output->is_outfile_append && compression_method != CompressionMethod::None)
```

Comment:
I would expect some compression methods to support appending new data to the end of the file, maybe we should check it and allow it for some of them?

Reply:
One caveat is this can corrupt data if user accidentally changes compression format while appending? Would you still prefer to allow formats with framing that supports appends to allow it?

Reply:
In general, the user can append the data in an incompatible format or with a different structure of fields and make the file unreadable anyways, so it's their responsibility to make sure everything is the same. But it is out of the scope of the original task anyways, so we can just merge the current version and deal with this later if this issue comes up.